projects/Mallard/src/helpers/weather-visibility.ts

```diff
@@ -0,0 +1,40 @@
+import { getSetting, storeSetting } from './settings'
+import { useState, useEffect } from 'react'
+
+type WeatherVisibility = 'shown' | 'hidden'
+type Data =
+    | { loading: true; value?: undefined; error?: undefined }
+    | { loading?: undefined; value: WeatherVisibility; error?: undefined }
+    | { loading?: undefined; value?: undefined; error: Error }
+let data: Data = { loading: true }
```

Comment:
Would it be better to manage this as state?

Reply:
No, because it is shared between views. This is a global state: as soon as we switch the setting to off in the settings, it hides the weather on the home page